src/main/frontend/handler/editor.cljs

```diff
@@ -778,41 +779,49 @@
        (outliner-core/delete-blocks! [block] {:children? children?})))))
 
 (defn- move-to-prev-block
-  [repo sibling-block format id value]
-  (when (and repo sibling-block)
-    (when-let [sibling-block-id (dom/attr sibling-block "blockid")]
-      (when-let [block (db/pull repo '[*] [:block/uuid (uuid sibling-block-id)])]
-        (let [original-content (util/trim-safe (:block/content block))
-              value' (-> (property/remove-built-in-properties format original-content)
-                         (drawer/remove-logbook))
-              new-value (str value' value)
-              tail-len (count value)
-              pos (max
-                   (if original-content
-                     (gobj/get (utf8/encode original-content) "length")
-                     0)
-                   0)]
-          (edit-block! block pos id
-                       {:custom-content new-value
-                        :tail-len tail-len
-                        :move-cursor? false})
-          {:prev-block block
-           :new-content new-value})))))
+  ([repo sibling-block format id value]
+   (move-to-prev-block repo sibling-block format id value true))
+  ([repo sibling-block format id value edit?]
+   (when (and repo sibling-block)
+     (when-let [sibling-block-id (dom/attr sibling-block "blockid")]
+       (when-let [block (db/pull repo '[*] [:block/uuid (uuid sibling-block-id)])]
+         (let [original-content (util/trim-safe (:block/content block))
+               value' (-> (property/remove-built-in-properties format original-content)
+                          (drawer/remove-logbook))
+               new-value (str value' value)
+               tail-len (count value)
+               pos (max
+                    (if original-content
+                      (gobj/get (utf8/encode original-content) "length")
+                      0)
+                    0)]
+           (when edit?
+             (edit-block! block pos id
+                          {:custom-content new-value
+                           :tail-len tail-len
+                           :move-cursor? false}))
+           {:prev-block block
+            :new-content new-value
+            :pos pos}))))))
 
 (declare save-block!)
 
+(defn- block-has-no-ref?
+  [block-or-uuid]
```

Comment:
Optional: It could be nice to keep this fn simpler and have it just take an entity id (eid) like db/pull or https://cljdoc.org/d/datascript/datascript/1.4.2/api/datascript.core#entity. This would mean passing in the `db/entity` arg

Reply:
sibling-block is retrieved through DOM, so only the UUID can be obtained. And block-has-no-ref? parameter does not uniformly use UUID because querying with UUID takes about 5 times longer then querying with eid (although the actual time consumption of both solutions is very short)